src/explore-education-statistics-admin/src/pages/release/datablocks/components/chart/__tests__/ChartBuilder.test.tsx

```diff
@@ -0,0 +1,321 @@
+import { testTableData } from '@admin/pages/release/datablocks/__data__/tableToolServiceData';
+import {
+  ChartBuilderForms,
+  ChartBuilderFormsContextProvider,
+} from '@admin/pages/release/datablocks/components/chart/contexts/ChartBuilderFormsContext';
+import { FullTableMeta } from '@common/modules/table-tool/types/fullTable';
+import mapFullTableMeta from '@common/modules/table-tool/utils/mapFullTableMeta';
+import mapFullTable from '@common/modules/table-tool/utils/mapFullTable';
+import { TableDataSubjectMeta } from '@common/services/tableBuilderService';
+import { render, screen, waitFor, within } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import noop from 'lodash/noop';
+import React from 'react';
+import ChartBuilder from '../ChartBuilder';
+
+describe('ChartBuilder', () => {
+  const testFormState: ChartBuilderForms = {
+    options: {
+      isValid: true,
+      submitCount: 0,
+      id: 'options',
+      title: 'Options',
+    },
+    dataSets: {
+      isValid: true,
+      submitCount: 0,
+      id: 'dataSets',
+      title: 'Data sets',
+    },
+  };
+
+  const testTable = mapFullTable(testTableData);
+  const testMeta: TableDataSubjectMeta = {
```

Comment:
This doesn't seem right - we're calling `mapFullTable` on some random other test data here when we're defining some suite-specific `testMeta` / `testSubjectMeta` that may not have any correlation to `testTableData`.

Would suggest something like the following instead:

```tsx
const testTableData: TableDataResponse = {
  subjectMeta: { ... },
  results: [ ... ]
};

const testTable = mapFullTable(testTableData);

<ChartBuilder
  results={testTable.results}
  meta={testTable.subjectMeta}
/>
```

Reply:
yeah don't know what I was doing there! Sorted now.